code/Controllers/CMSMain.php

```diff
@@ -828,6 +833,43 @@ public function getEditForm($id = null, $fields = null)
         }
     }
 
+    protected function getArchiveWarningMessage($record)
+    {
+        // Get all page's descendants
+        $record->collateDescendants(true, $descendants);
```

Comment:
Personal note: Investigate if this is a cause of significant CMS slowdown.

Reply:
@tractorcow @flamerohr @phalkunz Yes, it is. Which we've found out painfully by clients 9 months later in a large pre-production codebase, and is now causing us some pretty awkward discussions and team impact :/ https://github.com/silverstripe/silverstripe-admin/issues/571#issuecomment-416794591

Reply:
A good discussion for accuracy vs performance :)

Reply:
Hm, I think that's missing the point - in this case you can get accuracy *and* performance (by computing on demand when the info is actually used), but only if you think about this use case with performance in mind from the start